BatteryManagementSystem_Test.cpp

```diff
@@ -0,0 +1,37 @@
+/**************************************************************************************
+f* @file        : BatteryManagementSystem_test.cpp
+* @brief        : File contains necessary business logic to test the various parameters of BMS
+**************************************************************************************/
+
+#include <assert.h>
+#include <iostream>
+using namespace std;
+
+#include "BatteryManagementSystem_Factory.h"
+
+void testElectricVehicle_BMS(BatteryManagementSystem::BMSTestIF* bmsTestInstance)
+{
+	if(bmsTestInstance != NULL)
```

Comment:
The factory already produces a test-instance only. Why this further check here? Also it doesn't have an `else`... is it ok to ignore that condition?